### ISSUE TITLE
Remove user_meta related sniffs

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#614]
User meta is easily accessible on VIP Go, no reason to have the user_meta grouping in the RestrictedFunctions sniff:

https://github.com/Automattic/VIP-Coding-Standards/blob/6d3f5267bee6f0e7548aacc3881b2ac0bcd981e9/WordPressVIPMinimum/Sniffs/Functions/RestrictedFunctionsSniff.php#L132-L141

We silence it anyways on the WordPress-VIP-Go ruleset:

https://github.com/Automattic/VIP-Coding-Standards/blob/6d3f5267bee6f0e7548aacc3881b2ac0bcd981e9/WordPress-VIP-Go/ruleset.xml#L281-L290

The same could be said for the RestrictedVariables sniff:

https://github.com/Automattic/VIP-Coding-Standards/blob/6d3f5267bee6f0e7548aacc3881b2ac0bcd981e9/WordPressVIPMinimum/Sniffs/Variables/RestrictedVariablesSniff.php#L41-L47

Silenced on the same ruleset:

https://github.com/Automattic/VIP-Coding-Standards/blob/6d3f5267bee6f0e7548aacc3881b2ac0bcd981e9/WordPress-VIP-Go/ruleset.xml#L293-L295

**Comment by jrfnl** [comment]
Loosely related to #600.

The difference between #600 and these changes is that these are **_not_** breaking changes as no _sniffs_ are removed, only error codes (and groups).

This means that this change might make some inline ignore annotations _redundant_ and possibly make some values in custom properties in rulesets redundant, but other than that, it will not _break_ rulesets and it will not _break_ runs.

In other words, these changes are "allowed" to go into a next minor and don't _need_ to wait for the next major (though a choice can be made to let this wait for the next major anyway).

**Comment by jrfnl** [comment]
I have prepared the commits for this. @rebeccahum Let me know if you want me to pull these changes now or when we're preparing for v3.0.

**Comment by rebeccahum** [comment]
@jrfnl Please pull the changes now!
